### ISSUE TITLE
Better variable naming in Darknet models.

### ISSUE THREAD

**Issue by sebastian-sz** [keras-team/keras-cv#1297]
Currently when we inspect Darknet variables:
```python
from keras_cv.models import DarkNet51
m = DarkNet51(include_rescaling=False, include_top=True, classes=1000, input_shape=(224, 224, 3))
for v in m.variables:
    print(v.name)
```
we will get very unclear information about the variables:
```
batch_normalization_56/moving_variance:0
conv2d_57/kernel:0
batch_normalization_57/gamma:0
```
essentially a flat list of convs + BN. This makes inspecting variables or transfering weights very difficult if not impossible.

### The problem:
This will propagate on all Darknet based futere implementations: YoloV7 YoloV8, YoloX etc.

### The fix:
In [Darknet Conv Block](https://github.com/keras-team/keras-cv/blob/master/keras_cv/models/__internal__/darknet_utils.py#L48) the `name` variable is unused. One could change:
from
```python
    model_layers = [  # line 50
        layers.Conv2D(
            filters,
            kernel_size,
            strides,
            padding="same",
            use_bias=use_bias,
        ),
        layers.BatchNormalization(),
    ]
```

to

```python
    model_layers = [
        layers.Conv2D(
            filters,
            kernel_size,
            strides,
            padding="same",
            use_bias=use_bias,
            name=name+"_conv"
        ),
        layers.BatchNormalization(name=name+"_bn"),
    ]
```

After fix, the weights seem to be loading without issue on my machine and the variables have beautiful meaningful names:
```
dark5_conv4_conv/kernel:0
dark5_conv4_bn/gamma:0
dark5_conv4_bn/beta:0
dark5_conv4_bn/moving_mean:0
dark5_conv4_bn/moving_variance:0
```

**Comment by LukeWood** [comment]
Thanks Sebastian!  @quantumalaviya for more comments.

feel free to raise a PR addressing this

**Comment by quantumalaviya** [comment]
Sure, we can do this. Thanks!
> This makes inspecting variables or transfering weights very difficult if not impossible.

When you say "transferring weights" here, does it mean `model.load_weights()` is difficult or did you mean something else?

**Comment by sebastian-sz** [comment]
@quantumalaviya Thanks!

> does it mean model.load_weights()

Not really. Recently I played around with some Yolo implementations in Tensorflow and wanted to reuse existing darknet weights from this repo, but I gave up after noticing that it's a flat list of mostly integer names (with batchnorms appearing mostly before convs). 

I was thinking that if we have this option to add names to layers / variables it is much more readable and allows such weights reuse.
